src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs

```diff
@@ -675,5 +676,147 @@ public async Task PostAatfObligatedData_Always_SetsInternalBreadcrumbToViewRepor
+        /// <returns></returns>
+        [Fact]
+        public async Task PostAatfSentOnData_WithInvalidViewModel_ReturnsAatfSentOnDataViewModel()
+        {
+            // Arrange
+            var years = new List<int>() { 2019 };
+            A.CallTo(() => weeeClient.SendAsync(A<string>.Ignored, A<GetAatfReturnsActiveComplianceYears>.Ignored)).Returns(years);
+
+            // Act
+            controller.ModelState.AddModelError("Key", "Error");
+            var result = await controller.AatfSentOnData(new AatfSentOnDataViewModel());
+
+            // Assert
+            var viewResult = result as ViewResult;
+            Assert.NotNull(viewResult);
+            Assert.True(string.IsNullOrEmpty(viewResult.ViewName) || viewResult.ViewName == "AatfSentOnData");
+
+            var model = viewResult.Model as AatfSentOnDataViewModel;
+            Assert.NotNull(model);
+            Assert.Collection(model.ComplianceYears,
+                y1 => Assert.Equal("2019", y1.Text));
+        }
+
+        /// <summary>
+        /// This test ensures that the POST "AatfSentOnData" action with an invalid view model
+        /// returns a view with the ViewBag property "TriggerDownload" set to false.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task PostAatfSentOnData_WithInvalidViewModel_SetsTriggerDownloadToFalse()
+        {
+            // Act
+            controller.ModelState.AddModelError("Key", "Error");
+            var result = await controller.AatfSentOnData(new AatfSentOnDataViewModel());
+
+            // Assert
+            var viewResult = result as ViewResult;
+            Assert.NotNull(viewResult);
+            Assert.Equal(false, viewResult.ViewBag.TriggerDownload);
+        }
+
+        /// <summary>
+        /// This test ensures that the POST "AatfSentOnData" action with a valid view model
+        /// returns a view with the ViewBag property "TriggerDownload" set to true.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task PostAatfSentOnData_WithViewModel_SetsTriggerDownloadToTrue()
+        {
+            // Act
+            var result = await controller.AatfSentOnData(new AatfSentOnDataViewModel());
+
+            // Assert
+            var viewResult = result as ViewResult;
+            Assert.NotNull(viewResult);
+            Assert.Equal(true, viewResult.ViewBag.TriggerDownload);
+        }
+
+        /// <summary>
+        /// This test ensures that the POST "AatfSentOnData" action sets
+        /// the breadcrumb's internal activity to "View reports".
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task PostAatfSentOnData_Always_SetsInternalBreadcrumbToViewReports()
+        {
+            // Act
+            await controller.AatfSentOnData(A.Dummy<AatfSentOnDataViewModel>());
+
+            // Assert
+            Assert.Equal("View reports", breadcrumb.InternalActivity);
+        }
```

Comment:
Add test for DownloadAatfSentOnDataCsv to ensure the request is called

Reply:
Added test